src/hal/serial.h

```diff
@@ -0,0 +1,13 @@
+
+#ifndef __SERIAL_HAL_H__
+#define __SERIAL_HAL_H__
+
+//#define SERIAL_DEBUG
+
+#include "../hal/hal.h"
+
+int serialHalInit(void);
+int serialHalGetData(void);
```

Comment:
Please add //TODO: Doc on the tops of each of these -- we should add comments latter on each functions to define exactly what it does

Reply:
updated serial.cpp, serial.h, link.cpp and link.h with comments